### DISCUSSION TITLE
Is there a way to disable exposing OpenAPI Schema

### DISCUSSION THREAD

**Discussion by grieblius** [supabase/supabase#9464] [Questions] [answered]
Hello,

Is there a way to control Postgrest [openapi-mode](https://postgrest.org/en/stable/configuration.html#openapi-mode) option in Supabase?

Currently by default Supabase API (Postgrest) exposes DB schema as OpenAPI spec. Of course the OpenAPI output depends by role (`anon`, `authenticated`, etc), but I don't want to expose some tables/views or functions and detail it's structure. 
For example, there is some app admin functions and tables, and some `authenticated` users have ability to access and use it, but I do not want to expose it in public for other `authenticated` users. For some functions I could remove `authenticated` role and leave only `service_role` but not for all so it is not an option.
For me it would be enough to have OpenApi schema only locally or enable temporary in production on some cases.

I found this suggestion, however this does not work as it seems seems that Supabase uses environment variables instead of in-database configuration (at least  in local dev environment):
https://github.com/supabase/supabase/discussions/4547#discussioncomment-3487883

It would be good to control `openapi-mode` in Supabase panel API Settings.

Or maybe you can recommend other approach to not expose some tables and functions, e.g. using additional schemas? 

Thank you!

**Comment by grieblius** [comment, marked as answer]
Hello, 

seems I was wrong - Supabase PostgREST configuration is "in-database", so I succeeded to change `openapi-mode` myself.

It is possible to manage it by running query:

```sql
ALTER ROLE authenticator SET pgrst.openapi_mode TO 'disabled';
```

And then you need to reload PostgREST configuration by running:

```sql
NOTIFY pgrst, 'reload config'
```

**Reply by abhay187** [reply]
Thanks, @grieblius I was also worried about this being enabled by default.
This can be a significant security threat for someone who is not an expert in PostgreSQL. It can expose private tables that only the admin should access.
